lib/WarnCaseSensitiveModulesPlugin.js

```diff
@@ -2,27 +2,32 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var CaseSensitiveModulesWarning = require("./CaseSensitiveModulesWarning");
+"use strict";
 
-function WarnCaseSensitiveModulesPlugin() {}
-module.exports = WarnCaseSensitiveModulesPlugin;
+const CaseSensitiveModulesWarning = require("./CaseSensitiveModulesWarning");
 
-WarnCaseSensitiveModulesPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation) {
-		compilation.plugin("seal", function() {
-			var moduleWithoutCase = {};
-			this.modules.forEach(function(module) {
-				var ident = module.identifier().toLowerCase();
-				if(moduleWithoutCase["$" + ident]) {
-					moduleWithoutCase["$" + ident].push(module);
-				} else {
-					moduleWithoutCase["$" + ident] = [module];
-				}
-			}, this);
-			Object.keys(moduleWithoutCase).forEach(function(key) {
-				if(moduleWithoutCase[key].length > 1)
-					this.warnings.push(new CaseSensitiveModulesWarning(moduleWithoutCase[key]));
-			}, this);
+class WarnCaseSensitiveModulesPlugin {
+	apply(compiler) {
+		compiler.plugin("compilation", compilation => {
+			compilation.plugin("seal", function() {
+				const moduleWithoutCase = new Map();
+				this.modules.forEach(module => {
```

Comment:
Would it be cleaner to close over `compilation` instead of relying on `this` here? This would allow changing the `seal` callback into an arrow function.